### DISCUSSION TITLE
JSON Documents API

### DISCUSSION THREAD

**Discussion by c-nv-s** [alash3al/redix#39] [Ideas]
https://github.com/tidwall/summitdb#json-documents

summitdb has a decent JSON Documents Redis-Style API.

could it be possible to implement something like this to have responses back as JSON

**Comment by c-nv-s** [comment]
also... I haven't pulled the v5 image to test it out yet but from the new README it looks like you have removed a lot of the old functionality.
does this mean there is no more:

    SUBSCRIBE [<channel1> <channel2>], if there is no channel specified, it will be set to *
    PUBLISH <channel> <payload>
    WEBHOOKSET <channel> <httpurl>, register a http endpoint so it can be notified through JSON POST request with the channel updates, this command will return a reference ID so you can manage it later.
    WEBHOOKDEL <ID>, stops listening on a channel using the above reference ID.
    WEBSOCKETOPEN <channel>, opens a websocket endpoint and returns its id, so you can receive updates through ws://server.address:port/stream/ws/{generated_id_here}
    WEBSOCKETCLOSE <ID>, closes the specified websocket endpoint using the above generated id.

**Comment by alash3al** [comment]
Welcome, @c-nv-s ,

- Regarding JSON support, I was thinking about something similar to the mentioned example, and for sure everything could be supported, but let's define some reliable use cases that use json specific commands, i.e, why don't we just use `SET user:100 {JSON_PAYLOAD_HERE}`?
- Regarding the mentioned commands, yes I removed a lot of commands, but I plan to add them again but based on the new structure

**Reply by c-nv-s** [reply]
> * i.e, why don't we just use `SET user:100 {JSON_PAYLOAD_HERE}`?

yes this is fine. but then when you want to change just one specific field in the json you have to copy the whole thing, edit it the resubmit it?
unless you allow for doing ``SET user:100 .contactDetails.email "mr@something.tld"``
but I thought that is a different function signature so wouldn't be compatible with the typical Redis ``SET`` no?
whichever way you would do it, I think it would be a useful feature.

>*yes I removed a lot of commands, but I plan to add them again but based on the new structure

awesome I'll look forward to when those commands return because they were awesome!

**Reply by c-nv-s** [reply]
don't know how I've only just realized this, but this ``SET user:100 .contactDetails.email "mr@something.tld"`` is actually the function signature of ``HSET``

**Reply by c-nv-s** [reply]
so you could actually use ``HSET`` to map to ``jsonb`` column in postgres and that would give much more control over the json that is returned or stored
